### PR TITLE
refactor: Cypress SelectEntityByName

### PR DESCRIPTION
**Pull request by hetunandu**
Refactors SelectEntityByName to allow all types of selections but makes defining the Entity Type compulsory

### CODE REVIEW

File: app/client/cypress/e2e/Regression/Apps/CommunityIssues_Spec.ts [hunk 1]
```diff
@@ -11,6 +11,9 @@ import {
   assertHelper,
   draggableWidgets,
 } from "../../../support/Objects/ObjectsCore";
+import EditorNavigation, {
+  EntityType,
+} from "../../../support/Pages/EditorNavigation";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/Apps/CommunityIssues_Spec.ts:16]: Can this please be added as const object in Objects core & imported from that - similar to other page objects we have today?

Comment (hetunandu) [reply app/client/cypress/e2e/Regression/Apps/CommunityIssues_Spec.ts:16]: I do not think that is the right way to do it. Why is that a pattern today?

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/Apps/CommunityIssues_Spec.ts:16]: Oh, why is it not the right way please?

Comment (hetunandu) [reply app/client/cypress/e2e/Regression/Apps/CommunityIssues_Spec.ts:16]: The Object Registry is used to ensure a singleton is passed
```
static get CommunityTemplates(): CommunityTemplates {
    if (ObjectsRegistry.communityTemplates__ === undefined) {
      ObjectsRegistry.communityTemplates__ = new CommunityTemplates();
    }
    return ObjectsRegistry.communityTemplates__;
  }
```

That can be easily done in the export statement itself:
```
class EditorNavigation {
    ...
}

export default new EditorNavigation();
```


And the Object Core file is just to collect these exports to kind of improve import but that causes irregular patterns like
`_.entityExplorer.SelectEntityByName` or `entityExplorer.SelectEntityByName` or `this.entityExplorer.SelectEntityByName`


They cause unnecessary confusion on new users writing tests. It is better to always call the function via the class which will be same no matter where you import it.

```
import EditorNavigation from "...."

...



EditorNavigation.SelectEntityByName(...)

````

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/Apps/CommunityIssues_Spec.ts:16]: Ok, got it.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**